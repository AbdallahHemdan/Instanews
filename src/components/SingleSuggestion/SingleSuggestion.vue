<template>
  <div class="suggestion">
    <div class="suggestion__left">
      <img
        :src="user.userImg"
        :alt="user.username"
        class="suggestion__user-img"
        draggable="false"
      />

      <div class="suggestion__info">
        <div class="suggestion__username">{{ user.username }}</div>
        <span class="suggestion__relation">{{ user.suggestionBase }}</span>
      </div>
    </div>

    <div class="suggestion__right">
      <a :href="'/' + user.id" class="suggestion__follow-link">Follow</a>
    </div>
  </div>
</template>

<script>
export default {
  name: 'SingleSuggestion',
  props: {
    user: {
      type: Object,
      requires: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.suggestion {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: 0.5rem;

  &__left {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 6rem;
    margin-right: 134px;
  }

  &__relation {
    color: $dark-gray;
    font-size: 80%;
  }

  &__user-img {
    width: 32px;
    height: 32px;
    border-radius: 50%;
    margin-right: 1rem;
  }

  &__follow-link {
    text-decoration: none;
  }
}
</style>
