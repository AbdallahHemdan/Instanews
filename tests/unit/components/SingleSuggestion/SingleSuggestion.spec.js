import { shallowMount } from '@vue/test-utils';
import SingleSuggestion from './../../../../src/components/SingleSuggestion/SingleSuggestion.vue';

describe('SingleSuggestion.vue', () => {
  let cmp;
  const user = {
    userImg: require('./../../../../src/assets/Home-UserSuggestions/1.jpg'),
    username: 'sayedalesawy',
    suggestionBase: 'popular',
    id: '1',
  };

  beforeEach(() => {
    cmp = shallowMount(SingleSuggestion, {
      propsData: {
        user,
      },
    });
  });

  describe('Testing props', () => {
    it('should render user name correctly', () => {
      expect(cmp.find('.suggestion__username').text()).toBe(user.username);
    });

    it('should render user suggestion base correctly', () => {
      expect(cmp.find('.suggestion__relation').text()).toBe(user.suggestionBase);
    });

    it('should render user image correctly', () => {
      expect(cmp.find('.suggestion__user-img').attributes('src')).toBe(user.userImg);
    });

    it('should render user image alt text correctly', () => {
      expect(cmp.find('.suggestion__user-img').attributes('alt')).toBe(user.username);
    });

    it('should render user profile line correctly', () => {
      expect(cmp.find('.suggestion__follow-link').attributes('href')).toBe('/' + user.id);
    });
  });

  describe('Test inner text of elements', () => {
    it('should render follow link inner text correctly', () => {
      expect(cmp.find('.suggestion__follow-link').text()).toBe('Follow');
    });
  });

  describe('Snapshot testing', () => {
    it('should pass snapshot testing', () => {
      expect(cmp.element).toMatchSnapshot();
    });
  });
});
