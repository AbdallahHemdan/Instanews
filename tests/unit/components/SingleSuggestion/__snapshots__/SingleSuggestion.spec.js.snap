// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSuggestion.vue Snapshot testing should pass snapshot testing 1`] = `
<div
  class="suggestion"
>
  <div
    class="suggestion__left"
  >
    <img
      alt="sayedalesawy"
      class="suggestion__user-img"
      draggable="false"
      src=""
    />
     
    <div
      class="suggestion__info"
    >
      <div
        class="suggestion__username"
      >
        sayedalesawy
      </div>
       
      <span
        class="suggestion__relation"
      >
        popular
      </span>
    </div>
  </div>
   
  <div
    class="suggestion__right"
  >
    <a
      class="suggestion__follow-link"
      href="/1"
    >
      Follow
    </a>
  </div>
</div>
`;
